tree/ntuple/src/RNTupleMerger.cxx

```diff
@@ -533,6 +547,22 @@ CompareDescriptorStructure(const ROOT::RNTupleDescriptor &dst, const ROOT::RNTup
             }
          }
       }
+
+      // Require that subfields are compatible
+      const auto &srcLinks = field.fSrc->GetLinkIds();
+      const auto &dstLinks = field.fDst->GetLinkIds();
+      if (srcLinks.size() != dstLinks.size()) {
+         std::stringstream ss;
+         ss << "Field `" << field.fSrc->GetFieldName()
+            << "` has a different number of children than previously-seen field with the same name (old: "
+            << dstLinks.size() << ", new: " << srcLinks.size() << ")";
+         errors.push_back(ss.str());
+      } else
+         for (std::size_t linkIdx = 0, linkNum = srcLinks.size(); linkIdx < linkNum; ++linkIdx) {
+            const auto &srcSubfield = src.GetFieldDescriptor(srcLinks[linkIdx]);
+            const auto &dstSubfield = dst.GetFieldDescriptor(dstLinks[linkIdx]);
+            fieldsToCheck.push_back(RCommonField{srcSubfield, dstSubfield});
```

Comment:
It looks like you are implementing a workqueue algorithm, so `std::deque` might be more efficient and avoid collecting all fields.

Reply:
It is indeed a work queue algorithm, but `vector` works just fine in this case since I don't have to pop elements (incrementing the index is sufficient and faster)

Reply:
Sure, but we're talking about queues that hardly ever have more than ~1000 elements or so, each of whom is 2 pointers, so I don't think memory is much of a concern here. However if you disagree I can open a PR to change it.